File: solutions/problem_0010/java/LoginSimplesProject/src/dao/UserDao.java
package dao;

import model.User;

import java.util.List;

public interface UserDao {

    boolean insert(User user);

    List<User> findAll();

    User findByUsername(String username);

    User findByEmail(String email);

    boolean update(User user);

    boolean delete(User user);

    boolean deleteByUsername(String username);
}

File: solutions/problem_0010/kotlin/LoginSimplesProject/src/dao/UserDao.kt
package dao

import model.User

interface UserDao {

    fun insert(user: User): Boolean

    fun findAll(): List<User>

    fun findByUsername(username: String): User?

    fun findByEmail(email: String): User?

    fun update(user: User): Boolean

    fun delete(user: User): Boolean

    fun deleteByUsername(username: String): Boolean
}